awx/ui_next/src/components/MultiSelect/MultiSelect.jsx

```diff
@@ -127,37 +133,32 @@ class MultiSelect extends Component {
 
     this.setState({ chipItems: chips });
     onRemoveItem(item);
-
-    e.preventDefault();
   }
 
   render() {
     const { options } = this.props;
     const { chipItems, input, isExpanded } = this.state;
 
-    const list = options.map(option => (
-      <Fragment key={option.id}>
-        {option.name.includes(input) ? (
-          <DropdownItem
-            component="button"
-            isDisabled={chipItems.some(item => item.id === option.id)}
-            value={option.name}
-            onClick={e => {
-              this.handleClick(e, option);
-            }}
-          >
-            {option.name}
-          </DropdownItem>
-        ) : null}
-      </Fragment>
-    ));
+    const list = options
+      .filter(filteredOption => filteredOption.name.includes(input))
+      .map(option => (
+        <DropdownItem
+          key={option.id}
+          component="button"
+          isDisabled={chipItems.some(item => item.id === option.id)}
+          value={option.name}
+          onClick={this.handleClick}
+        >
+          {option.name}
+        </DropdownItem>
+      ));
 
     const chips = (
       <ChipGroup>
         {chipItems &&
           chipItems.map(item => (
             <Chip
-              key={item.id}
+              key={item.id || Math.random()}
```

Comment:
We should probably avoid using Math.random() to generate keys. React docs include Math.random() as an example of how to generate an unstable key: https://reactjs.org/docs/reconciliation.html#tradeoffs. Since the label name must always be unique, what do you think about using that as the key instead?

Reply:
yes I think we can make that happen.